modules/detection/lidar_driver.py

```diff
@@ -0,0 +1,270 @@
+        self._packet_payload_size = 0
+        self._packet_size = 0
+        self._packet_data = []
+
+    def _create_crc(self, data):
+        """
+        Create a CRC-16-CCITT 0x1021 hash of the specified data.
+        """
+        crc = 0
+
+        for i in data:
+            code = crc >> 8
+            code ^= int(i)
+            code ^= code >> 4
+            crc = crc << 8
+            crc ^= code
+            code = code << 5
+            crc ^= code
+            code = code << 7
+            crc ^= code
+            crc &= 0xFFFF
+
+        return crc
+
+    def _build_packet(self, command, write, data=None):
+        """
+        Create raw bytes for a packet.
+        """
+        if data is None:
+            data = []
+
+        payload_length = 1 + len(data)
+        flags = (payload_length << 6) | (write & 0x1)
+        packet_bytes = [0xAA, flags & 0xFF, (flags >> 8) & 0xFF, command]
+        packet_bytes.extend(data)
+        crc = self._create_crc(packet_bytes)
+        packet_bytes.append(crc & 0xFF)
+        packet_bytes.append((crc >> 8) & 0xFF)
+
+        return bytearray(packet_bytes)
+
+    def _parse_packet(self, byte):
+        """
+        Check for packet in byte stream.
+        """
+        if self._packet_parse_state == 0:
+            if byte == 0xAA:
+                self._packet_parse_state = 1
+                self._packet_data = [0xAA]
+
+        elif self._packet_parse_state == 1:
+            self._packet_parse_state = 2
+            self._packet_data.append(byte)
+
+        elif self._packet_parse_state == 2:
+            self._packet_parse_state = 3
+            self._packet_data.append(byte)
+            self._packet_payload_size = (self._packet_data[1] | (self._packet_data[2] << 8)) >> 6
+            self._packet_payload_size += 2
+            self._packet_size = 3
+
+            if self._packet_payload_size > 1019:
+                self._packet_parse_state = 0
+
+        elif self._packet_parse_state == 3:
+            self._packet_data.append(byte)
+            self._packet_size += 1
+            self._packet_payload_size -= 1
+
+            if self._packet_payload_size == 0:
+                self._packet_parse_state = 0
+                crc = self._packet_data[self._packet_size - 2] | (
+                    self._packet_data[self._packet_size - 1] << 8
+                )
+                verify_crc = self._create_crc(self._packet_data[0:-2])
+
+                if crc == verify_crc:
+                    return True
+
+        return False
+
+    def _wait_for_packet(self, port, command, timeout=1):
+        """
+        Wait (up to timeout) for a packet of the specified command to be received.
+        """
+        self._packet_parse_state = 0
+        self._packet_data = []
+        self._packet_payload_size = 0
+        self._packet_size = 0
+
+        end_time = time.time() + timeout
+
+        while True:
+            if time.time() >= end_time:
+                return None
```

Comment:
change this to `while time.time() < end_time` instead, and have the return at the end (you also don't need to say `return None`, it can just be `return`)